writer.go

```diff
@@ -106,6 +106,12 @@ type WriterConfig struct {
 	// whether the messages were written to kafka.
 	Async bool
 
+	// CompressionCodec set the codec to be used to compress Kafka messages.
+	CompressionCodec int8
+
+	// CompressionLevel set the compression level if the codec supports it.
+	CompressionLevel int
```

Comment:
Is this still necessary? How about having a `CompressionCodec` field instead so the program can set the compression algorithm to whatever it pleases.